src/qibo/models/encodings.py

```diff
@@ -156,6 +156,8 @@ def unary_encoder(data, architecture: str = "tree"):
     circuit_rbs, _ = _generate_rbs_pairs(nqubits, architecture=architecture)
     circuit += circuit_rbs
 
+    if data.__class__.__name__ == "Tensor":
+        data = data.detach().numpy()
```

Comment:
This shouldn't be the solution because the same input data might be something that needs gradients later on.